hazelcast/src/main/java/com/hazelcast/sql/impl/client/SqlClientService.java

```diff
@@ -93,23 +91,23 @@ public SqlResult execute(@Nonnull SqlStatement statement) {
                 statement.getExpectedResultType().getId(),
                 id,
                 skipUpdateStatistics
-            );
+        );
 
-            SqlClientResult res = new SqlClientResult(
+        SqlClientResult res = new SqlClientResult(
                 this,
                 connection,
                 id,
                 statement.getCursorBufferSize()
-            );
-
-            ClientInvocationFuture future = invokeAsync(requestMessage, connection);
-
-            future.whenComplete(withTryCatch(logger,
-                    (message, error) -> handleExecuteResponse(connection, res, message, error))).get();
+        );
 
+        try {
+            ClientMessage message = invoke(requestMessage, connection);
+            handleExecuteResponse(res, message);
             return res;
         } catch (Exception e) {
-            throw rethrow(e, connection);
+            RuntimeException error = rethrow(e, connection);
```

Comment:
if the connection is closed in between the error thrown in handleExecuteResponse and the rethrow here; we will report to user a `Cluster topology changed while a query was executed` error, while in fact it was a different error. Should we fix this?

Reply:
We can add this 
![image](https://user-images.githubusercontent.com/32355782/134675754-02f101a6-d88f-4c6e-a1f6-2dfb727c06b4.png)

@hazelcast/sql

Reply:
Isn't it better to throw the original cause? What is the benefit of ignoring it?

Reply:
I suppose that the error can be misleading in this case, but I don't exactly know which. Maybe some operation timeout or something else. The point is that whatever happened, if the connection is down, that's the major issue.

Unfortunately we don't assert the `memberConnection` error in any test...